source/PlayerInfo.cpp

```diff
@@ -3170,6 +3233,18 @@ void PlayerInfo::SelectShip(const shared_ptr<Ship> &ship, bool *first)
 
 
 
+// When we remove a ship, forget its stored ID.
+void PlayerInfo::ForgetShip(const Ship &oldShip)
+{
+	const EsUuid &id = oldShip.UUID();
+	auto shipToForget = find_if(giftedShips.begin(), giftedShips.end(),
+						[&id](const std::pair<const string, EsUuid> &shipId) { return id == shipId.second; });
+	if(shipToForget != giftedShips.end())
+		giftedShips.erase(shipToForget->first);
```

Comment:
`map::erase()` can just take an iterator, which you've already got, no need to go get the key.
```suggestion
		giftedShips.erase(shipToForget);
```